stubs/pynput/pynput/mouse/_base.pyi

```diff
@@ -8,41 +8,41 @@ from typing_extensions import Self
 from pynput._util import AbstractListener
 
 class Button(enum.Enum):
-    unknown: int
-    left: int
-    middle: int
-    right: int
+    unknown = 0
+    left = 1
+    middle = 2
+    right = 3
     if sys.platform == "linux":
-        button8: int
-        button9: int
-        button10: int
-        button11: int
-        button12: int
-        button13: int
-        button14: int
-        button15: int
-        button16: int
-        button17: int
-        button18: int
-        button19: int
-        button20: int
-        button21: int
-        button22: int
-        button23: int
-        button24: int
-        button25: int
-        button26: int
-        button27: int
-        button28: int
-        button29: int
-        button30: int
-        scroll_down: int
-        scroll_left: int
-        scroll_right: int
-        scroll_up: int
+        button8 = ...
```

Comment:
This change leads to a regression in how mypy infers the type of the value of this enum member. On `main`:
```
% MYPYPATH=stubs/pynput mypy -c "import pynput.mouse._base; reveal_type(pynput.mouse._base.Button.button8.value)" --platform linux
<string>:1: note: Revealed type is "builtins.int"
```

With this PR:
```
MYPYPATH=stubs/pynput mypy -c "import pynput.mouse._base; reveal_type(pynput.mouse._base.Button.button8.value)" --platform linux
<string>:1: note: Revealed type is "types.EllipsisType"
```

Reply:
I'd prefer not to wait for a bug fix in mypy 1.10. That will take another several weeks. 

Do you know where we can find the values for these enum members? IIRC, it wasn't obvious from the sources. I suppose we could use introspection to determine their values, but I'd need help from someone on a Linux and Windows machine to do this.

Reply:
Here are the values on linux with `pynput==1.7.6`:

```
>>> from pynput.mouse import Button
>>> list(Button)
[<Button.unknown: None>, <Button.left: 1>, <Button.middle: 2>, <Button.right: 3>, <Button.scroll_up: 4>, <Button.scroll_down: 5>, <Button.scroll_left: 6>, <Button.scroll_right: 7>, <Button.button8: 8>, <Button.button9: 9>, <Button.button10: 10>, <Button.button11: 11>, <Button.button12: 12>, <Button.button13: 13>, <Button.button14: 14>, <Button.button15: 15>, <Button.button16: 16>, <Button.button17: 17>, <Button.button18: 18>, <Button.button19: 19>, <Button.button20: 20>, <Button.button21: 21>, <Button.button22: 22>, <Button.button23: 23>, <Button.button24: 24>, <Button.button25: 25>, <Button.button26: 26>, <Button.button27: 27>, <Button.button28: 28>, <Button.button29: 29>, <Button.button30: 30>]
```

The stub is a bit misleading. These are actually defined in `pynput.mouse._xorg`, not in `pynput.mouse._base`. There is another, different `Button` enum in `pynput.mouse._base`, but that one lacks most of the buttons:

```
>>> Button.__module__
'pynput.mouse._xorg'
>>> from pynput.mouse._base import Button
>>> Button.__module__
'pynput.mouse._base'
>>> list(Button)
[<Button.unknown: 0>, <Button.left: 1>, <Button.middle: 2>, <Button.right: 3>]
>>> Button.button8
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.11/enum.py", line 789, in __getattr__
    raise AttributeError(name) from None
AttributeError: button8
```

Reply:
I don't see these in the source code at all (https://github.com/moses-palmer/pynput/blob/12acf84dc0f721d91a957da65311497acb664933/lib/pynput/mouse/_base.py#L33), are they added dynamically somewhere?

I tried installing pynput on a Linux server but it fails to import; apparently it requires a running X server. I don't have time right now to try to cajole it, but I might take another look later.

Reply:
> I'd prefer not to wait for a bug fix in mypy 1.10. That will take another several weeks.

Somewhat adjacent to the point you're making, but what does pyright infer for the value of the `button8` member here @erictraut? I'm not sure this does qualify as a mypy bug really; I don't know how it could be expected to know that the value of the member really is an `int` here

Reply:
The `...` token means `Any` in this context, so that's what pyright assumes. Mypy should do the same.

Reply:
I see. That would still be a regression, though, since currently mypy is able to infer `int`.

Reply:
Following https://github.com/python/typeshed/pull/11299/commits/bd170fab90094ce6f711af36db556299234a4c58, here's what mypy now infers:

```
% MYPYPATH=stubs/pynput mypy -c "import pynput.mouse._base; reveal_type(pynput.mouse._base.Button.button8.value)" --platform linux
<string>:1: note: Revealed type is "Literal[0]?"
```

According to @Akuli's comment in https://github.com/python/typeshed/pull/11299#discussion_r1561336972, that's also incorrect, since the value of the `button8` member is `8`. Why don't you just use `8` for this member, and `13` for the `button13` member, etc., according to the values Akuli listed in https://github.com/python/typeshed/pull/11299#discussion_r1561336972?

Reply:
Ah, I missed @Akuli's post above. I added the linux-specific values. There are still two windows-specific values where the values are still unknown.

Reply:
They are tuples under Windows. The x1 and x2 values simply don't exist in the `_base.Button` enum, but does in `_win32.Button`:
https://github.com/moses-palmer/pynput/blob/12acf84dc0f721d91a957da65311497acb664933/lib/pynput/mouse/_win32.py#L48-L56

Reply:
Yes, we might want to do a round of review on these stubs to make them follow the implementation more closely, that would make it easier to verify correctness. That's out of scope for this PR, though.

Reply:
Yeah, I appreciate the desire to make these values perfectly match the implementation, but I'd appreciate it if we could unblock this PR (which is itself blocking a bunch of things) and then do a separate pass to refine these types. Are you OK with that?